agent/test-agent-cli/design/RUNBOOK.md

```diff
@@ -0,0 +1,195 @@
+test-agent-cli send-result -o fail -p 0 -f 1 -s 0
+
+if test; then
+    break
+fi
+done
+fi
+
+# Save the test result after consuming all the retries,  that will contain
+# the info how many test cases failed, passed, skipped
+test-agent-cli send-result -o pass -p 1 -f 0 -s 0
+
+# Set the Task_state as completed. This command will also create the tar file of the results
+# This will change the resource status in testsys to passed/failed, but Pod state will remain running if keep_running has been set as true
+# To change the pod state to complete set keep_running as false(testsys set test_name --keep-running false)
+test-agent-cli terminate --results-dir results_directory
+
+# Setting the task_status and resource status as error if some error encounters
+# test-agent-cli send-error error_encountered
+```
+
+2. Create a docker file, that will be used to create image that will be running on the Test Pod. Following is an example of the Docker file that provides all the required resources to run the Bash script along with the Command Line Interface in a single pod. In this docker file make sure to mark entry point as the Test Bash script created in Step 1.
+
+```shell
+# syntax=docker/dockerfile:1.1.3-experimental
+
+ARG BUILDER_IMAGE
+FROM ${BUILDER_IMAGE} as build
+
+ARG ARCH
+USER root
+
+# We need these environment variables set for building the `openssl-sys` crate
+
+ENV PKG_CONFIG_PATH=/${ARCH}-bottlerocket-linux-musl/sys-root/usr/lib/pkgconfig
+ENV PKG_CONFIG_ALLOW_CROSS=1
+ENV CARGO_HOME=/src/.cargo
+ENV OPENSSL_STATIC=true
+ENV CARGO_HOME=/src/.cargo
+ADD ./ /src/
+WORKDIR /src/agent/test-agent-cli
+RUN --mount=type=cache,mode=0777,target=/src/target 
+cargo install --locked --offline --target ${ARCH}-bottlerocket-linux-musl --path .  --root ./
+
+FROM public.ecr.aws/amazonlinux/amazonlinux:2
+RUN yum update -y 
+&& yum install -y jq 
+&& yum clean all
+COPY --from=build /src/agent/test-agent-cli/bin/test-agent-cli /usr/local/bin/
+COPY --from=build /src/agent/test-agent-cli/examples/example_test_agent_cli/example-test.sh ./
+
+ENTRYPOINT ["/bin/bash", "./example-test.sh"]
+```
+
+3. Create a docker image consists of the Command Line Interface, Bash Script test and all other resources required to run the test as running the following command from Bottlerocket Test System root folder.(Before running this make sure to add the details to create this image using docker build in makefile(https://github.com/bottlerocket-os/bottlerocket-test-system/blob/develop/Makefile))
+   make example-test-agent-cli
+4. Create a controller image using the following command from Bottlerocket Test System root folder.
+   `make controlller`
+5. Create any Resource Agent image(if current test depend on any Resource) in the same way as we do in existing process.
+6. Create a kind cluster as *kind create cluster --name testsys*. I prefer to first delete the cluster if any exists using *kind delete cluster --name testsys* .
```

Comment:
We can use "`" instead of *

Reply:
I'd remove "I prefer to first delete the cluster if any exists using *kind delete cluster --name testsys* ."